mypy_json_report.py

```diff
@@ -229,6 +242,94 @@ class DiffReport:
     num_fixed_errors: int
 
 
+class _ChangeReportWriter(Protocol):
+    def write_report(self, diff: DiffReport) -> None:
+        ...
+
+
+class DefaultChangeReportWriter:
+    """Writes an error summary without color."""
+
+    def __init__(self, _write: Callable[[str], Any] = sys.stdout.write) -> None:
+        self.write = _write
+
+    def write_report(self, diff: DiffReport) -> None:
+        new_errors = "\n".join(diff.error_lines)
+        self.write(new_errors + "\n")
+        self.write(f"Fixed errors: {diff.num_fixed_errors}\n")
+        self.write(f"New errors: {diff.num_new_errors}\n")
+        self.write(f"Total errors: {diff.total_errors}\n")
+
+
+_RESET = "\033[0m"
+_BOLD = "\033[1m"
+_BOLD_RED = "\033[31;1m"
+_GREEN = "\033[32m"
+_YELLOW = "\033[33m"
+_BLUE = "\033[34m"
+
+
+class ColorChangeReportWriter:
+    """
+    Writes an error summary in color.
+
+    Inspired by the FancyFormatter in mypy.util.
+
+    Ref: https://github.com/python/mypy/blob/f9e8e0bda5cfbb54d6a8f9e482aa25da28a1a635/mypy/util.py#L761
+    """
+
+    def __init__(self, _write: Callable[[str], Any] = sys.stdout.write) -> None:
+        self.write = _write
+
+    def write_report(self, diff: DiffReport) -> None:
+        new_errors = "\n".join([self._format_line(line) for line in diff.error_lines])
+        self.write(new_errors + "\n")
+
+        fixed_color = _BOLD_RED if diff.num_fixed_errors else _GREEN
```

Comment:
I did have that in my first iteration, but I came around to doing it this way because I thought that this would be more appropriate when seen in a CI (or pre-commit) context. This matches the fact that the return-code will be non-zero when changes to the ratchet-file are made.

(Although I do see the argument that fixing errors is a good thing, and would usually be green.)

Reply:
Perhaps making the text yellow when there are fixed errors would be better?

Reply:
I can do that!